pkg/apis/apps/v1/zz_generated.defaults.go

```diff
@@ -107,7 +109,9 @@ func SetObjectDefaults_DaemonSet(in *v1.DaemonSet) {
 		corev1.SetDefaults_Container(a)
 		for j := range a.Ports {
 			b := &a.Ports[j]
-			corev1.SetDefaults_ContainerPort(b)
+			if reflect.ValueOf(b.Protocol).IsZero() {
```

Comment:
curious how expensive this is... did we check the impact of adding many reflect calls to the defaulting path instead of generating `x == ...` checks for scalars?

Reply:
We have a benchmark here https://github.com/kubernetes/gengo/pull/192#discussion_r520955114.

The Protocol string field is the only field that we are looking to apply this defaulting to for 1.20 and we already have a follow up task to use direct comparisons for primitive types.

Reply:
It was the first thing I asked too. It appears not to allocate, and it's slow but we only do it for this single field. @Jefftree will fix before we add more fields.